src/test/java/org/isf/ward/test/Tests.java

```diff
@@ -8,354 +8,219 @@
 import org.isf.utils.db.DbJpaUtil;
 import org.isf.utils.exception.OHException;
 import org.isf.ward.model.Ward;
+import org.isf.ward.service.WardIoOperationRepository;
 import org.isf.ward.service.WardIoOperations;
-import org.junit.After;
-import org.junit.AfterClass;
-import org.junit.Before;
-import org.junit.BeforeClass;
-import org.junit.Test;
+import org.junit.*;
 import org.junit.runner.RunWith;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.junit4.SpringRunner;
 
 @RunWith(SpringRunner.class)
 @ContextConfiguration(locations = { "classpath:applicationContext.xml" })
-public class Tests  
-{
-	private static DbJpaUtil jpa;
+public class Tests {
 	private static TestWard testWard;
-	private static TestWardContext testWardContext;
 
     @Autowired
     WardIoOperations wardIoOperation;
+    @Autowired
+    WardIoOperationRepository wardIoOperationRepository;
 	
 	@BeforeClass
-    public static void setUpClass()  
-    {
-    	jpa = new DbJpaUtil();
+    public static void setUpClass() {
     	testWard = new TestWard();
-    	testWardContext = new TestWardContext();
-    	
-        return;
     }
 
     @Before
-    public void setUp() throws OHException
-    {
-        jpa.open();
-        
-        _saveContext();
-		
-		return;
-    }
-        
-    @After
-    public void tearDown() throws Exception 
-    {
-        _restoreContext();   
-        
-        jpa.flush();
-        jpa.close();
-                
-        return;
+    public void setUp() {
+		wardIoOperationRepository.deleteAll();
     }
     
     @AfterClass
-    public static void tearDownClass() throws OHException 
-    {    	
+    public static void tearDownClass() throws OHException {
     	return;
     }
-	
```

Comment:
This part has been changed compared to others, for consistency with other tests we need all these methods:

```
    @BeforeClass
    public static void setUpClass() {
		jpa = new DbJpaUtil();
    	testWard = new TestWard();
    	testWardContext = new TestWardContext();
    }

    @Before
    public void setUp() throws OHException {
    	jpa.open();
        
        _saveContext();
		
		return;
    }
    
    @After
    public void tearDown() throws Exception 
    {
        _restoreContext();   
        
        jpa.flush();
        jpa.close();
                
        return;
    }
    
    @AfterClass
    public static void tearDownClass() throws OHException {
    	
    	testWard = null;
    	testWardContext = null;
    	
    	return;
    }

...

private void _saveContext() throws OHException 
    {	
		testWardContext.saveAll(jpa);
		
        return;
    }
	
	private void _restoreContext() throws OHException 
    {
		testWardContext.deleteNews(jpa);
        
        return;
    }
```

Reply:
Are you sure we need this? Now we start fresh in memory db each time we run tests so context stuff is not needed anymore. Repository is cleaned up in line 35 
public void setUp() {
		wardIoOperationRepository.deleteAll();
so tests are independent, and now we have much less code to maintain (just one line to cleanup db before test)

Reply:
I know... but so why that context stuff is maintained in all other classes?

When I run all the tests (not only `Ward `or `MedicaStockWard ` packages) I get the error once reached `testIoNewMovementWard`:

```
ERROR - Violación de indice de Unicidad ó Clave primaria: "PRIMARY_KEY_28 ON PUBLIC.WARD(WRD_ID_A) VALUES (CAST('X' AS VARCHAR_IGNORECASE), 59)"
Unique index or primary key violation: "PRIMARY_KEY_28 ON PUBLIC.WARD(WRD_ID_A) VALUES (CAST('X' AS VARCHAR_IGNORECASE), 59)"; SQL statement:
insert into WARD (WRD_ACTIVE, WRD_CREATED_BY, WRD_CREATED_DATE, WRD_LAST_MODIFIED_BY, WRD_LAST_MODIFIED_DATE, WRD_NBEDS, WRD_NAME, WRD_NDOC, WRD_EMAIL, WRD_FAX, WRD_IS_FEMALE, WRD_IS_MALE, WRD_IS_PHARMACY, WRD_LOCK, WRD_NQUA_NURS, WRD_TELE, WRD_ID_A) values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?) [23505-190]
[18/jul/2020 23:35:51] [:] ERROR - Violación de indice de Unicidad ó Clave primaria: "PRIMARY_KEY_28 ON PUBLIC.WARD(WRD_ID_A) VALUES (CAST('X' AS VARCHAR_IGNORECASE), 59)"
Unique index or primary key violation: "PRIMARY_KEY_28 ON PUBLIC.WARD(WRD_ID_A) VALUES (CAST('X' AS VARCHAR_IGNORECASE), 59)"; SQL statement:
insert into WARD (WRD_ACTIVE, WRD_CREATED_BY, WRD_CREATED_DATE, WRD_LAST_MODIFIED_BY, WRD_LAST_MODIFIED_DATE, WRD_NBEDS, WRD_NAME, WRD_NDOC, WRD_EMAIL, WRD_FAX, WRD_IS_FEMALE, WRD_IS_MALE, WRD_IS_PHARMACY, WRD_LOCK, WRD_NQUA_NURS, WRD_TELE, WRD_ID_A) values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?) [23505-190]
javax.persistence.PersistenceException: org.hibernate.exception.ConstraintViolationException: could not execute statement
```

The strange thing is that in CI it doesn't happen...

Reply:
ERRATA CORRIGE: it fails only in my Eclipse (Idkw) but in Maven build all goes smooth.

Reply:
Ok than I can restore it and adjust tests setup in separate pull request what do you think? Of course if you agree with this kind of change but in my opinion possibility to remove this context stuff from tests is one of benefits having in memory db for tests